distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -101,6 +102,27 @@ make_column_ref_by_name_selector table display:Display=..Always add_text:Boolean
     values = if columns_first then col_names + constants + expression else constants + expression + col_names
     Single_Choice values=values display=display
 
+## PRIVATE
+   Make a column reference by name selector filtered by type.
+make_column_ref_by_name_selector_for_type table display:Display=..Always value_type:Value_Type add_nothing:Boolean=False =
+    is_all = include_all value_type
+    include_number = is_all || value_type.is_numeric
+    include_text = is_all || value_type.is_text
+    include_boolean = is_all || value_type.is_boolean
+    include_date = is_all || value_type.is_date
+    include_date_time = is_all || value_type.is_date_time
+    include_time = is_all || value_type.is_time
+
+    constants = make_any_selector add_text=include_text add_number=include_number add_boolean=include_boolean add_date=include_date add_time=include_time add_date_time=include_date_time add_nothing=add_nothing . values
+    expression = if table.is_nothing then [] else [Option "<Expression>" "(expr '["+table.column_names.first+"]')"]
+
+    col_names = if table.is_nothing then [] else
+        names = if is_all then table.column_names else (table.select_columns [..By_Type value_type] . column_names)
```

Comment:
Should we also include Mixed columns there?

i.e. if I have
```
table.filter "int_column" (..Equal _)
```
I guess I should still be able to check if it's equal to a `"mixed"` column - as that can contain some Integer values as well.

So I'd suggest
```suggestion
        names = if is_all then table.column_names else (table.select_columns [..By_Type value_type, ..By_Type Value_Type.Mixed] . column_names)
```
to include both columns of current type but also mixed columns.